bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareRegistry.java

```diff
@@ -54,107 +55,113 @@
+                        "Unexpected exception occurred for firmware provider {} while getting firmware with version {} for thing {}",
+                        firmwareProvider.getClass().getSimpleName(), firmwareVersion, thing, e);
             }
         }
 
         return null;
     }
 
     /**
-     * Returns the latest firmware for the given thing type UID using the locale provided by the {@link LocaleProvider}.
+     * Returns the latest firmware for the given thing, using the locale provided by the {@link LocaleProvider}.
      *
-     * @param thingTypeUID the thing type UID (must not be null)
+     * @param thing the thing for which the firmwares are to be retrieved (not null)
      * @return the corresponding latest firmware or null if no firmware was found
-     * @throws NullPointerException if given thing type UID is null
+     * @throws NullPointerException if given thing is null
      */
-    public Firmware getLatestFirmware(ThingTypeUID thingTypeUID) {
-        return getLatestFirmware(thingTypeUID, localeProvider.getLocale());
+    public Firmware getLatestFirmware(Thing thing) {
+        return getLatestFirmware(thing, localeProvider.getLocale());
     }
 
     /**
-     * Returns the latest firmware for the given thing type UID and locale.
+     * Returns the latest firmware for the given thing and locale.
      *
-     * @param thingTypeUID the thing type UID (must not be null)
+     * @param thing the thing for which the firmwares are to be retrieved (not null)
      * @param locale the locale to be used (if null then the locale provided by the {@link LocaleProvider} is used)
      * @return the corresponding latest firmware or null if no firmware was found
-     * @throws NullPointerException if given thing type UID is null
+     * @throws NullPointerException if given thing is null
      */
-    public Firmware getLatestFirmware(ThingTypeUID thingTypeUID, Locale locale) {
+    public Firmware getLatestFirmware(Thing thing, Locale locale) {
         Locale loc = locale != null ? locale : localeProvider.getLocale();
-        return getFirmwares((thingTypeUID), loc).stream().findFirst().orElse(null);
+        return getFirmwares(thing, loc).stream().findFirst().orElse(null);
     }
 
     /**
-     * Returns the collection of available firmwares for the given thing type UID using the locale provided by the
+     * Returns the collection of available firmwares for the given thing using the locale provided by the
      * {@link LocaleProvider}. The collection is sorted in descending order, i.e. the latest firmware will be the first
      * element in the collection.
      *
-     * @param thingTypeUID the thing type UID for which the firmwares are to be provided (not null)
-     * @return the collection of available firmwares for the given thing type UID (not null)
-     * @throws NullPointerException if given thing type UID is null
+     * @param thing - the thing for which the firmwares are to be retrieved (not null)
```

Comment:
Is there a ```-```to be removed?

Reply:
Yes, all these issues were reported by the SCAT tool of ESH, so I will fix them with the next update on the PR.